firebase-ml-modeldownloader/src/androidTest/java/com/google/firebase/ml/modeldownloader/testGetModelLocal.java

```diff
@@ -0,0 +1,272 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.firebase.ml.modeldownloader;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import android.os.ParcelFileDescriptor;
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.runner.AndroidJUnit4;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.ml.modeldownloader.internal.ModelFileManager;
+import com.google.firebase.ml.modeldownloader.internal.SharedPreferencesUtil;
+import java.io.File;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Integration tests for Get Model, download type LOCAL_MODEL. Only use getLocalModel* models for
+ * this test.
+ */
+@RunWith(AndroidJUnit4.class)
+public class testGetModelLocal {
+  private FirebaseModelDownloader firebaseModelDownloader;
+  private static final String MODEL_NAME_LOCAL = "getLocalModel";
+  private static final String MODEL_NAME_LOCAL_2 = "getLocalModel2";
+  private static final String MODEL_HASH = "origHash324";
+  private final CustomModel SETUP_LOADED_LOCAL_MODEL =
+      new CustomModel(MODEL_NAME_LOCAL, MODEL_HASH, 100, 0);
+
+  private FirebaseApp app;
+  private File firstDeviceModelFile;
+  private File firstLoadTempModelFile;
+  private SharedPreferencesUtil sharedPreferencesUtil;
+
+  @Before
+  public void before() throws ExecutionException, InterruptedException {
+    if (firebaseModelDownloader == null) {
```

Comment:
iiuc it's always going to be `null` since `@Before` runs before each test on a newly created instance of this class.

Reply:
thanks - fixed